app/services/image_validation.py

```diff
@@ -0,0 +1,211 @@
+            logger.warning(f'Invalid content type: {content_type} for file {filename}')
+            raise HTTPException(
+                status_code=400,
+                detail=f'Invalid content type "{content_type}". Allowed types: {", ".join(settings.image_allowed_types)}',
+            )
+
+        # Step 3: Read file content into memory
+        # Note: For very large files, consider streaming validation
+        file_content = await file.read()
+        file_size = len(file_content)
+
+        # Validate file size
+        max_size_bytes = settings.image_max_size_mb * 1024 * 1024
+        if file_size > max_size_bytes:
+            size_mb = file_size / (1024 * 1024)
+            logger.warning(f'File too large: {size_mb:.2f} MB for file {filename}')
+            raise HTTPException(
+                status_code=400,
+                detail=f'Image size {size_mb:.2f} MB exceeds maximum allowed {settings.image_max_size_mb} MB',
+            )
+
+        if file_size == 0:
+            logger.warning(f'Empty file uploaded: {filename}')
+            raise HTTPException(status_code=400, detail='Uploaded file is empty.')
+
+        # Step 4: Validate magic bytes (actual file content)
+        magic_bytes_valid = _validate_magic_bytes(file_content, content_type)
+        if not magic_bytes_valid:
+            logger.warning(f'Magic bytes validation failed for {filename} (claimed type: {content_type})')
+            raise HTTPException(
+                status_code=400,
+                detail='File content does not match declared type. Possible file corruption or spoofing.',
+            )
+
+        # Step 5: Validate image can be opened and get dimensions
+        try:
+            image = Image.open(io.BytesIO(file_content))
+            width, height = image.size
+            image_format = image.format  # PIL detected format
+
+            logger.info(f'Image opened successfully: {filename} ({width}x{height}, format: {image_format})')
+
+        except UnidentifiedImageError:
+            logger.warning(f'Cannot identify image format for {filename}')
+            raise HTTPException(status_code=400, detail='Invalid or corrupted image file.')
+        except Exception as e:
+            logger.error(f'Error opening image {filename}: {str(e)}')
+            raise HTTPException(status_code=400, detail='Cannot process image. File may be corrupted.')
+
+        # Step 6: Validate resolution
+        if width > settings.image_max_resolution or height > settings.image_max_resolution:
+            logger.warning(f'Resolution too high: {width}x{height} for file {filename}')
+            raise HTTPException(
+                status_code=400,
+                detail=f'Image resolution {width}x{height} exceeds maximum allowed {settings.image_max_resolution}x{settings.image_max_resolution}',
+            )
+
+        # Step 7: Reset file pointer for potential future reads
+        await file.seek(0)
```

Comment:
Resetting file pointer without checking support may fail for some UploadFile implementations. Add a try-except to handle seek errors gracefully.
```
try:
    await file.seek(0)
except Exception:
    pass
```